app/controllers/api/databases.php

```diff
@@ -2900,6 +3158,37 @@ function () use ($dbForProject, $privateCollectionId, $document, $data) {
             throw new Exception(Exception::DOCUMENT_INVALID_STRUCTURE, $exception->getMessage());
         }
 
+        // Add $collectionId and $databaseId for all documents
+        $setIds = function (Document $collection, Document $document) use (&$setIds, $dbForProject, $database) {
+            $document->setAttribute('$databaseId', $database->getId());
+            $document->setAttribute('$collectionId', $collection->getId());
+
+            $relationships = \array_filter(
```

Comment:
Instead of having relationships be a list, it might be better to make it a hash table where the key is the attribute key so that a for loop isn't needed later.

Reply:
Either that, or iterate over all relationships instead of all attributes in the document.